op-node/rollup/derive/channel_test.go

```diff
@@ -253,17 +257,33 @@ func TestBatchReader(t *testing.T) {
 			}
 			require.NoError(t, err)
 
-			// read the batch data
+			// read the 1st batch data
 			batchData, err := reader()
 			require.NoError(t, err)
 			require.NotNil(t, batchData)
 			if tc.algo.IsBrotli() {
 				// special case because reader doesn't decode level
-				batchDataInput.ComprAlgo = Brotli
+				batchDataInput1.ComprAlgo = Brotli
+			} else {
+				batchDataInput1.ComprAlgo = tc.algo
+			}
+			require.Equal(t, batchDataInput1, batchData)
+
+			// read the 2nd batch data
+			batchData, err = reader()
+			require.NoError(t, err)
+			require.NotNil(t, batchData)
+			if tc.algo.IsBrotli() {
+				// special case because reader doesn't decode level
+				batchDataInput2.ComprAlgo = Brotli
 			} else {
-				batchDataInput.ComprAlgo = tc.algo
+				batchDataInput2.ComprAlgo = tc.algo
 			}
-			require.Equal(t, batchDataInput, batchData)
+			require.Equal(t, batchDataInput2, batchData)
```

Comment:
If we have slices, we can run this as a loop, instead of copy-pasting.

Reply:
Done in a11dbba.